game/dino_jump_main.py

```diff
@@ -0,0 +1,394 @@
+            filled_location = np.bitwise_or(left_edge, set_location)
+            is_hit = np.sum(np.bitwise_and(left_edge, set_location))
+        else:
+            filled_location = np.bitwise_or(right_edge, set_location)
+            is_hit = np.sum(np.bitwise_and(right_edge, set_location))
+        self.game_state_array[y : y + height, min_width:max_width] = filled_location
+
+        # If a hit is detected, we end the game
+        if is_hit:
+            self.game_over = True
+
+    def draw_to_screen(self):
+
+        # Use GRID_SIZE to increase the software viewing panel
+        draw_array = np.repeat(
+            np.repeat(self.game_state_array, self.GRID_SIZE, axis=0),
+            self.GRID_SIZE,
+            axis=1,
+        )
+
+        # For each pixel in the game board, draw them in black and white to the screen
+        for row in range(draw_array.shape[0]):
+            for col in range(draw_array.shape[1]):
+                if draw_array[row, col] == 1:
+                    pygame.draw.rect(
+                        self.screen,
+                        self.BLOCK_COLOR,
+                        (col, row, 1, 1),
+                    )
+
+    def check_start_game(self):
+        for event in pygame.event.get():
+            if event.type == pygame.KEYDOWN and event.key == pygame.K_SPACE:
+                self.start_game = True
+                self.dino.is_jumping = True
+                self.dino.update_jumping_location()
+
+    def check_user_input(self):
+        for event in pygame.event.get():
+            if event.type == pygame.QUIT:
+                pygame.quit()
+                sys.exit()
+
+            if event.type == pygame.KEYDOWN:
+                if event.key == pygame.K_SPACE:
+                    self.dino.is_jumping = True
+                if event.key == pygame.K_DOWN:
+                    self.dino.is_ducking = True
+
+            if event.type == pygame.KEYUP:
+                if event.key == pygame.K_SPACE:
+                    self.dino.is_jumping = False
+                if event.key == pygame.K_DOWN:
+                    self.dino.is_ducking = False
+
+    def draw_all(self):
+        self.screen.fill(game.BG_COLOR)
+        self.game_state_array = np.zeros((game.HEIGHT, game.WIDTH))
+        for obs in self.all_obstacles.curr_obs:
+            self.update_board_pixels(obs)
+        self.update_board_pixels(self.dino)
+        self.draw_to_screen()
+        pygame.display.flip()
+        self.clk.tick(self.FPS)
+
+    def check_close_panel(
+        self,
+    ):  # JOSH checks for another space bar press to close the whole program
+        for event in pygame.event.get():
+            if event.type == pygame.KEYDOWN and event.key == pygame.K_SPACE:
+                self.close_all = True
+
+
+#############################################################################
+# SINGLE OBSTACLE MODULE
+#############################################################################
+
+
+class SingleObstacle:

```

Comment:
nit: I would rename this class to just `Obstacle`

Reply:
Renamed!